src/core/public/application/types.ts

```diff
@@ -289,12 +330,16 @@ export type AppSearchDeepLink = {
       path: string;
       /** Optional array of links that are 'underneath' this section in the hierarchy */
       searchDeepLinks?: AppSearchDeepLink[];
+      /** Optional meta containing keywords to match with in deep links search */
+      meta?: AppMeta;
     }
   | {
       /** Optional path to access this section. Omit if this part of the hierarchy does not have a page URL. */
       path?: string;
       /** Array links that are 'underneath' this section in this hierarchy. */
       searchDeepLinks: AppSearchDeepLink[];
+      /** Meta containing keywords to match with in deep links search */
+      meta?: AppMeta;
```

Comment:
Strictly speaking, if no `path` is provided, we don't have a UI to navigate to and adding keywords in this case would not be beneficial. I left `meta` in as optional though, to keep the type declaration cleaner.